app/src/main/java/com/github/sdp/mediato/api/themoviedb/TheMovieDB.java

```diff
@@ -0,0 +1,113 @@
+package com.github.sdp.mediato.api.themoviedb;
+
+import com.github.sdp.mediato.api.API;
+import com.github.sdp.mediato.api.themoviedb.models.PagedResult;
+import com.github.sdp.mediato.api.themoviedb.models.TMDBMovie;
+import com.github.sdp.mediato.util.AdapterRetrofitCallback;
+
+import org.checkerframework.checker.units.qual.A;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class TheMovieDB implements API<TMDBMovie> {
+    private final TheMovieDBAPI api;
+    private final String apikey;
+    private final HashMap<String, ArrayList<TMDBMovie>> searchCache;
+
+    private final HashMap<String, Integer> searchPage;
+    private ArrayList<TMDBMovie> trendingCache;
+    private int trendingPage;
+
+    public TheMovieDB(String serverUrl, String apikey){
+        this.trendingCache = new ArrayList<>();
+        this.trendingPage = 0;
+        this.searchCache = new HashMap<>();
+        this.searchPage = new HashMap<>();
+        this.apikey = apikey;
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl(serverUrl)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build();
+        this.api = retrofit.create(TheMovieDBAPI.class);
+    }
+
+    @Override
+    public CompletableFuture<TMDBMovie> searchItem(String s) {
```

Comment:
Shoudn't we check length of string argument ?

Reply:
nice catch